receiver/dockerstatsreceiver/receiver.go

```diff
@@ -111,12 +116,264 @@ func (r *receiver) scrape(ctx context.Context) (pmetric.Metrics, error) {
+		"total_rss_huge":            r.mb.RecordContainerMemoryTotalRssHugeDataPoint,
+		"dirty":                     r.mb.RecordContainerMemoryDirtyDataPoint,
+		"total_dirty":               r.mb.RecordContainerMemoryTotalDirtyDataPoint,
+		"writeback":                 r.mb.RecordContainerMemoryWritebackDataPoint,
+		"total_writeback":           r.mb.RecordContainerMemoryTotalWritebackDataPoint,
+		"mapped_file":               r.mb.RecordContainerMemoryMappedFileDataPoint,
+		"total_mapped_file":         r.mb.RecordContainerMemoryTotalMappedFileDataPoint,
+		"pgpgin":                    r.mb.RecordContainerMemoryPgpginDataPoint,
+		"total_pgpgin":              r.mb.RecordContainerMemoryTotalPgpginDataPoint,
+		"pgpgout":                   r.mb.RecordContainerMemoryPgpgoutDataPoint,
+		"total_pgpgout":             r.mb.RecordContainerMemoryTotalPgpgoutDataPoint,
+		"swap":                      r.mb.RecordContainerMemorySwapDataPoint,
+		"total_swap":                r.mb.RecordContainerMemoryTotalSwapDataPoint,
+		"pgfault":                   r.mb.RecordContainerMemoryPgfaultDataPoint,
+		"total_pgfault":             r.mb.RecordContainerMemoryTotalPgfaultDataPoint,
+		"pgmajfault":                r.mb.RecordContainerMemoryPgmajfaultDataPoint,
+		"total_pgmajfault":          r.mb.RecordContainerMemoryTotalPgmajfaultDataPoint,
+		"inactive_anon":             r.mb.RecordContainerMemoryInactiveAnonDataPoint,
+		"total_inactive_anon":       r.mb.RecordContainerMemoryTotalInactiveAnonDataPoint,
+		"active_anon":               r.mb.RecordContainerMemoryActiveAnonDataPoint,
+		"total_active_anon":         r.mb.RecordContainerMemoryTotalActiveAnonDataPoint,
+		"inactive_file":             r.mb.RecordContainerMemoryInactiveFileDataPoint,
+		"total_inactive_file":       r.mb.RecordContainerMemoryTotalInactiveFileDataPoint,
+		"active_file":               r.mb.RecordContainerMemoryActiveFileDataPoint,
+		"total_active_file":         r.mb.RecordContainerMemoryTotalActiveFileDataPoint,
+		"unevictable":               r.mb.RecordContainerMemoryUnevictableDataPoint,
+		"total_unevictable":         r.mb.RecordContainerMemoryTotalUnevictableDataPoint,
+		"hierarchical_memory_limit": r.mb.RecordContainerMemoryHierarchicalMemoryLimitDataPoint,
+		"hierarchical_memsw_limit":  r.mb.RecordContainerMemoryHierarchicalMemswLimitDataPoint,
+	}
+
+	for name, val := range memoryStats.Stats {
+		if recorder, ok := recorders[name]; ok {
+			recorder(now, int64(val))
+		}
+	}
+}
+
+type blkioRecorder func(now pcommon.Timestamp, val int64, devMaj string, devMin string)
+
+type blkioMapper struct {
+	opToRecorderMap map[string]blkioRecorder
+	entries         []dtypes.BlkioStatEntry
+}
+
+func (r *receiver) recordBlkioMetrics(now pcommon.Timestamp, blkioStats *dtypes.BlkioStats) {
+	// These maps can be avoided once the operation is changed to an attribute instead of being in the metric name
+	for _, blkioRecorder := range []blkioMapper{
```

Comment:
One of the breaking changes I am suggesting to make in a later PR would be to change `read, write, etc.` into an attribute, so we wouldn't need so many entries here. Also way less generated code+docs.

Reply:
Is there any reason why these can't happen now?

Reply:
I don't want to change the existing metrics in this PR. I want to prove this implementation is equivalent output to the existing implementation, and then we can work on it from there. 

I've listed the improvements in this comment in the parent issue: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/9794#issuecomment-1182616779